core/templates/dev/head/collection_editor/CollectionEditor.js

```diff
@@ -0,0 +1,213 @@
+      if (skillName && expId) {
+        var collectionNode = _getCollectionNodeForExplorationId(expId);
+        var prerequisiteSkills = collectionNode.prerequisite_skills;
+        if (!_arrayContainsCaseInsensitive(prerequisiteSkills, skillName)) {
+          prerequisiteSkills.push(skillName);
+          return true;
+        }
+      }
+      return false;
+    };
+
+    // TODO(mgowano): Handle cases where deleting a pre-requisite skill results
+    // to unreachable explorations.
+    $scope.deletePrerequisiteSkill = function(skillIdx, expId) {
+      if (expId) {
+        var collectionNode = _getCollectionNodeForExplorationId(expId);
+        collectionNode.prerequisite_skills.splice(skillIdx, 1);
+        return true;
+      }
+      return false;
+    };
+
+    $scope.addAcquiredSkill = function(skillName, expId) {
+      if (skillName && expId) {
+        var collectionNode = _getCollectionNodeForExplorationId(expId);
+        var acquiredSkills = collectionNode.acquired_skills;
+        if (!_arrayContainsCaseInsensitive(acquiredSkills, skillName)) {
+          acquiredSkills.push(skillName);
+          return true;
+        }
+      }
+      return false;
+    };
+
+    // TODO(mgowano): Handle cases where deleting an acquired skill results
+    // to unreachable explorations.
+    $scope.deleteAcquiredSkill = function(skillIdx, expId) {
+      if (expId) {
+        var collectionNode = _getCollectionNodeForExplorationId(expId);
+        collectionNode.acquired_skills.splice(skillIdx, 1);
+        return true;
+      }
+      return false;
+    };
+
+    // TODO(mgowano): Handle a case where deleting an exploration will
+    // result into unreachable exploration. At the moment this is only
+    // handled when a change is committed to the backend.
+    $scope.deleteCollectionNode = function(expId) {
+      if (expId) {
+        for (var i = 0; i < $scope.collection.nodes.length; i++) {
+          var collectionNode = $scope.collection.nodes[i];
+          if (collectionNode.exploration_id == expId) {
+            $scope.collection.nodes.splice(i, 1);
+            return true;
+          }
+        }
+      }
+      return false;
+    };
+
+    // TODO(mgowano): Set all values for the 'exploration' key in newNode
+    $scope.addCollectionNode = function(expId) {
+      if (expId) {
+        if (!_getCollectionNodeForExplorationId(expId)) {
+          var newNode = {
+            exploration_id: expId,
+            acquired_skills: [],
+            prerequisite_skills: [],
+            exploration: {
+              id: expId
+            }
+          };
+          $scope.collection.nodes.push(newNode);
+          return true;
+        } else {
+          warningsData.addWarning(
+            'Exploration with id ' + expId + ' is already added');
+        }
+      }
+      return false;
+    };
+
+    $scope.setCollectionTitle = function(newTitle) {
+      if (newTitle) {
```

Comment:
The policies of these functions have changed, so I don't think your comment applies anymore (and should be addressed now that a warning is issued to the user for many of these actions).